Source/Core/Core/HW/DVD/DVDThread.cpp

```diff
@@ -358,6 +363,37 @@ static void FinishRead(u64 id, s64 cycles_late)
                                        buffer);
 }
 
+static void PatchReadRequest(std::vector<u8> &buffer, u64 offset) {
+  if(!s_disc_patches)
+    return;
+
+  u64 end_offset = offset + buffer.size();
+
+  // Scan through patches which might overlap our current read request and apply them
+  for (auto it = s_disc_patches->begin(); it != s_disc_patches->end(); it++) {
```

Comment:
Would it be possible to use `upper_bound` instead of looping through all patches? Kinda like we do in this function in the file system code: https://github.com/dolphin-emu/dolphin/blob/372b8552ce7c94c026e2309c0b9089440bdae672/Source/Core/DiscIO/FileSystemGCWii.cpp#L299

Reply:
This isn't looping through all patches since maps are sorted and there's a "return" in the loop once reaching an element that's too big.

However it feels like you could indeed micro-optimize this with some chosen .lower_bound() / .upper_bound() which are O(lg N). Given that this map is tiny, I don't know if it's really worth it.

Reply:
Yeah, in the case of what we're going to use this for now, the list is tiny and it doesn't really matter if we use a loop. I was more thinking of if we want to use this for more things in the future (phire mentioned Riivolution on IRC as a potential thing), but I guess that would require bigger changes anyway since patches can't be larger than 4 bytes in the current code.

Reply:
I had that optimisation, but the logic was incorrect so I patched it out for testing.

Will reverse the search order and add it back in later.

Reply:
I've re-written this loop to search backwards from upper_bound.